### PR TITLE
Add field mapping config

### PR DESCRIPTION
**Pull request by blueo**
This adds the ability to define aliases that map client-facing fields to internal field names or map existing field names to different names.

A new config variable `api_field_mapping` has been added that takes an array of the shape:
```php
[
  'fieldAlias' => 'DataObjectFieldName',
];
```

And mapping has been added to the handers and formatters. 

Feedback is much appreciated.

### CODE REVIEW

File: src/DataFormatter.php [hunk 1]
```diff
@@ -356,4 +356,87 @@ public function convertStringToArray($strData)
...
+     * Convert an array of aliased field names to their Dataobject field name
+     *
+     * @param string $className
+     * @param string[] $fields
+     * @return string[]
+     */
+    public function getRealFields($className, $fields)
+    {
+        $apiMapping = Config::inst()->get($className, 'api_field_mapping');
+        if (is_array($apiMapping) && is_array($fields)) {
```
Comment (robbieaverill) [inline src/DataFormatter.php]: Should we simplify this by using `$this-> getApiMapping()`?

File: src/RestfulServer.php [hunk 2]
```diff
@@ -13,6 +13,7 @@
 use SilverStripe\Security\Member;
 use SilverStripe\Security\Security;
 use SilverStripe\CMS\Model\SiteTree;
+use TractorCow\Fluent\Task\ConvertTranslatableTask\Exception;
```
Comment (robbieaverill) [inline src/RestfulServer.php]: Mistake?

File: tests/unit/RestfulServerTest.php [hunk 3]
```diff
@@ -112,6 +113,18 @@ public function testAuthenticatedGET()
         unset($_SERVER['PHP_AUTH_PW']);
     }
 
+    public function testGETWithFieldAlias()
+    {
+        Config::inst()->update(RestfulServerTestAuthorRating::class, 'api_field_mapping', ['rate' => 'Rating']);
```
Comment (robbieaverill) [inline tests/unit/RestfulServerTest.php]:
```diff
- Config::inst()->update(
+ Config::modify()->set(
```

Also apply to the other test method uses

**Comment by blueo** [comment]
Thanks @robbieaverill, changes made - thanks for picking up those mistakes

**Comment by robbieaverill** [comment]
## Wait to merge

I'd like to wait until CWP 2.0.0 is released before we merge this

**Pull request merged**